### PR TITLE
feat(Core/Mmaps): Handle different slopes in  mmaps and Add some more input parameters to improve

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/b6e205f7ef32f7b82644043d82a808e67cc9acb2
https://github.com/TrinityCore/TrinityCore/commit/3947e4cb57d4c6adb78cd3cb52718bb88add91fc
https://github.com/TrinityCore/TrinityCore/commit/995a443da219ec773febd7dd29d18f3cefaa1f3b
https://github.com/TrinityCore/TrinityCore/commit/7b561373a892c74870a21cf31687df4afd554495

## Tests Performed:

- Need Test again
- Re extract needed
- Part of multithreading mmaps

## How to Test the Changes:

1.Please read the source description
2. test with this pr https://github.com/azerothcore/azerothcore-wotlk/pull/10922
3.make sure blink, warrior charge works fine Especially in the arena

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by IntelligentQuantum** [comment]
@pklloveyou Can you test this pr please

**Comment by pklloveyou** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/10841
https://github.com/azerothcore/azerothcore-wotlk/pull/10921
https://github.com/azerothcore/azerothcore-wotlk/pull/10922
Working properly on my server.
It will be safer to perform the next operation after merging.

**Review by Winfidonarleyan — APPROVED** [review]

File: src/tools/mmaps_generator/MapBuilder.cpp [hunk 1]
```diff
@@ -633,16 +637,25 @@ namespace MMAP
...
+
+                /* we want to have triangles with slope less than walkableSlopeAngleNotSteep (<= 55) to have NAV_AREA_GROUND
+                 * and with slope between walkableSlopeAngleNotSteep and walkableSlopeAngle (55 < .. <= 70) to have NAV_AREA_GROUND_STEEP.
+                 * we achieve this using recast API: memset everything to NAV_AREA_GROUND_STEEP, call rcClearUnwalkableTriangles with 70 so
+                 * any area above that will get RC_NULL_AREA (unwalkable), then call rcMarkWalkableTriangles with 55 to set NAV_AREA_GROUND
+                 * on anything below 55 . Players and idle Creatures can use NAV_AREA_GROUND, while Creatures in combat can use NAV_AREA_GROUND_STEEP.
+                 */
                 unsigned char* triFlags = new unsigned char[tTriCount];
-                memset(triFlags, NAV_GROUND, tTriCount * sizeof(unsigned char));
+                memset(triFlags, NAV_AREA_GROUND_STEEP, tTriCount*sizeof(unsigned char));
```
Comment (Winfidonarleyan) [inline src/tools/mmaps_generator/MapBuilder.cpp:648]:
```suggestion
                memset(triFlags, NAV_AREA_GROUND_STEEP, tTriCount * sizeof(unsigned char));
```

File: src/tools/mmaps_generator/MapBuilder.cpp [hunk 2]
```diff
@@ -740,8 +753,15 @@ namespace MMAP
...
         for (int i = 0; i < iv.polyMesh->npolys; ++i)
-            if (iv.polyMesh->areas[i] & RC_WALKABLE_AREA)
-                iv.polyMesh->flags[i] = iv.polyMesh->areas[i];
+        {
+            if (uint8 area = iv.polyMesh->areas[i] & NAV_AREA_ALL_MASK)
+            {
+                if (area >= NAV_AREA_MIN_VALUE)
+                    iv.polyMesh->flags[i] = 1 << (NAV_AREA_MAX_VALUE - area);
+                else
+                    iv.polyMesh->flags[i] = NAV_GROUND; // TODO: these will be dynamic in future
```
Comment (Winfidonarleyan) [inline src/tools/mmaps_generator/MapBuilder.cpp:762]: brackets

File: src/tools/mmaps_generator/PathCommon.h [hunk 3]
```diff
@@ -105,7 +105,7 @@ namespace MMAP
             errno = 0;
             if ((dp = readdir(dirp)) != nullptr)
             {
-                if (matchWildcardFilter(filter.c_str(), dp->d_name))
+                if (strcmp(dp->d_name, ".") != 0 && strcmp(dp->d_name, "..") != 0 && matchWildcardFilter(filter.c_str(), dp->d_name))
```
Comment (Winfidonarleyan) [inline src/tools/mmaps_generator/PathCommon.h:108]: you can use this
https://github.com/azerothcore/azerothcore-wotlk/blob/3961e0839c8106d04e6ace425c1160771b1aa558/src/common/Utilities/Util.h#L395

File: src/tools/mmaps_generator/PathGenerator.cpp [hunk 4]
```diff
@@ -84,11 +100,23 @@ bool handleArgs(int argc, char** argv,
...
             else
                 printf("invalid option for '--maxAngle', using default\n");
         }
+        else if (strcmp(argv[i], "--maxAngleNotSteep") == 0)
+        {
+            param = argv[++i];
+            if (!param)
+                return false;
+
+            float maxangle = atof(param);
```
Comment (Winfidonarleyan) [inline src/tools/mmaps_generator/PathGenerator.cpp:114]: You can use this
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/common/Utilities/StringConvert.h

File: src/tools/mmaps_generator/PathGenerator.cpp [hunk 5]
```diff
@@ -277,7 +322,11 @@ int main(int argc, char** argv)
     if (!checkDirectories(debugOutput))
         return silent ? -3 : finish("Press ENTER to close...", -3);
 
-    MapBuilder builder(maxAngle, skipLiquid, skipContinents, skipJunkMaps,
+    _liquidTypes = LoadLiquid();
+    if (_liquidTypes.empty())
+        return silent ? -5 : finish("Failed to load LiquidType.dbc", -5);
```
Comment (Winfidonarleyan) [inline src/tools/mmaps_generator/PathGenerator.cpp:327]: brackets

File: src/tools/mmaps_generator/PathCommon.h (hunk 3 above)
Comment (IntelligentQuantum) [reply src/tools/mmaps_generator/PathCommon.h:108]: These changes are made after completing the maps.

File: src/tools/mmaps_generator/PathGenerator.cpp (hunk 4 above)
Comment (IntelligentQuantum) [reply src/tools/mmaps_generator/PathGenerator.cpp:114]: These changes are made after completing the maps.

**Comment by Nefertumm** [comment]
Conflicts

**Comment by IntelligentQuantum** [comment]
@Nefertumm Done

**Comment by Kitzunu** [comment]
conflict

**Review by Kitzunu — APPROVED** [review]

**Comment by Helias** [comment]
is the AC data have been updated even in the dashboard bash tools?

**Pull request merged**